### DISCUSSION TITLE
i do not see a postgres table although session works

### DISCUSSION THREAD

**Discussion by HosMercury** [maxcountryman/tower-sessions#111] [Q&A] [answered]
```
////////////////// SESSION //////////////////////////////
    let session_store = PostgresStore::new(pool.clone());
    session_store.migrate().await.unwrap();

    let session_service = ServiceBuilder::new()
        .layer(HandleErrorLayer::new(|_: BoxError| async {
            StatusCode::BAD_REQUEST
        }))
        .layer(
            SessionManagerLayer::new(session_store)
                .with_secure(false)
                .with_name("sid")
                .with_expiry(Expiry::OnInactivity(Duration::seconds(1000))),
        );
```

I do not see any db tables
although the session get / set works ?

**Comment by and-reas-se** [comment, marked as answer]
The tables are created in a separate schema named tower_sessions. You need to switch to that schema in your postgres client to see the tables.

I agree that this is a bit confusing, and should probably be documented better. I think it would make more sense to default to using the default schema when the user doesn't specify one, but that would be a breaking change now.

**Reply by HosMercury** [reply]
no there is not

**Comment by HosMercury** [comment]
i do not understand here ?
should I migrate it manually ?

i am sure this table is not exists

**Reply by maxcountryman** [reply]
Have you followed the docs and examples?

You've given us very little information to help with.

**Reply by HosMercury** [reply]
i have followed this example 
https://github.com/maxcountryman/tower-sessions/blob/main/examples/postgres-store.rs

and copied the code from 
as I wrote above 

the session works , there is no errors 

but I do not see any tables 
![postgres client](https://github.com/maxcountryman/tower-sessions/assets/8302755/c1beb430-d5d0-4498-8b78-2174a0b28e41)

**Reply by maxcountryman** [reply]
As you've been told previously, you need to switch to the `tower_sessions` schema: you are using the `public` schema. 
<img width="241" alt="Screenshot 2023-12-04 at 11 00 05" src="https://github.com/maxcountryman/tower-sessions/assets/74351/da91ec6c-1316-4c7a-905a-4dc4ce737a1a">

**Reply by HosMercury** [reply]
aha 
ok now i saw it 
this is very confusing tbh

**Reply by maxcountryman** [reply]
It's common for a library to use its own schema: doing so isolates the library's schema and helps to ensure that there aren't accidental namespace clashes.

It's a bit beyond the scope of this crate to document Postgres features and the best practices around them, so I would certainly encourage you to do more research if you're unclear or find it confusing.

**Reply by HosMercury** [reply]
yes thanx
